### PR TITLE
[DO NOT MERGE] Adding DataStore JS lazy loading docs

### PR DESCRIPTION
**Pull request by renebrandel**
_Description of changes:_
- Updated all predicates to use new format
- Changed code relational models sections
- Expanded the many-to-many section

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/datastore/js/relational/query-many-snippet.mdx [hunk 1]
```diff
@@ -1,48 +1,48 @@
...
-            "id": "2cbfdd83-8353-4b0e-ae63-8f7d004c728f",
-            "username": "Nadia"
-        }
-    }
-]
+Here we've first created a new `Post` instance and a new `User` instance. Then, saved those instances to a new instance of our join model `PostEditor`.
+
+### Query
+
+To query many-to-many relationships, query either model with a nested predicate of the other model's primary key.
```
Comment (abdallahshaban557) [inline src/fragments/lib/datastore/js/relational/query-many-snippet.mdx:26]: I think the other model's here is a big unclear.

**Review by abdallahshaban557 — COMMENTED** [review]
Left 1 piece of feedback

**Review by svidgen — CHANGES_REQUESTED** [review]
Need to address statement around "implicit `and`" groups, which are no longer a thing.

Also missing predicate updates for sync expressions on `Syncing data to cloud` page. Missing predicate updates for Angular and Vue snippets.

File: src/fragments/lib/datastore/js/relational/query-many-snippet.mdx (hunk 1 above)
Comment (renebrandel) [reply src/fragments/lib/datastore/js/relational/query-many-snippet.mdx:26]: do you have a suggestion?

File: src/fragments/lib/datastore/js/data-access/query-predicate-multiple-snippet.mdx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 When using multiple conditions, there is an implicit `and` defined to mirror the [GraphQL Transformer condition statements](/cli/graphql/data-modeling). For example with multiple conditions:
```
Comment (renebrandel) [inline src/fragments/lib/datastore/js/data-access/query-predicate-multiple-snippet.mdx:1]:
```suggestion
When using multiple conditions, you can wrap the predicates with the `and` operator. For example with multiple conditions:
```

**Review by svidgen — DISMISSED** [review]

**Review by stocaaro — APPROVED** [review]

File: src/fragments/lib/datastore/js/relational/query-many-snippet.mdx (hunk 1 above)
Comment (renebrandel) [reply src/fragments/lib/datastore/js/relational/query-many-snippet.mdx:26]:
```suggestion
To query many-to-many relationships, use a nested predicate filter the target model by the source model's id:
```

File: src/fragments/lib/datastore/js/relational/query-many-snippet.mdx [hunk 3]
```diff
@@ -1,48 +1,48 @@
...
-            "id": "2cbfdd83-8353-4b0e-ae63-8f7d004c728f",
-            "username": "Nadia"
-        }
-    }
-]
+Here we've first created a new `Post` instance and a new `User` instance. Then, saved those instances to a new instance of our join model `PostEditor`.
+
+### Query
+
+To query many-to-many relationships, use a nested predicate filter the target model by the source model's id:
```
Comment (renebrandel) [inline src/fragments/lib/datastore/js/relational/query-many-snippet.mdx:26]:
```suggestion
To query many-to-many relationships, use a nested predicate to filter the target model by the source model's id:
```

**Review by katiegoines — APPROVED** [review]

**Review by abdallahshaban557 — APPROVED** [review]
LGTM!

**Review by svidgen — APPROVED** [review]

**Pull request merged**